Tasks/Gradle/CodeAnalysis/Common/CodeAnalysisResultPublisher.ts

```diff
@@ -101,28 +101,31 @@ export class CodeAnalysisResultPublisher {
     private createSummaryContent(): string {
 
         var buildSummaryLines: string[] = [];
-        var resultsGroupedByTool: AnalysisResult[][] = this.groupBy(this.analysisResults, (o: AnalysisResult) => { return o.toolName; });
+        var resultsGroupedByTool: AnalysisResult[][] =
+            this.groupBy(this.analysisResults, (o: AnalysisResult) => { return o.originatingTool.toolName; });
 
         for (var resultGroup of resultsGroupedByTool) {
             var summaryLine = this.createSummaryLine(resultGroup);
-            buildSummaryLines.push(summaryLine);
+            if (summaryLine != null) {
+                buildSummaryLines.push(summaryLine);
+            }
         }
 
-        tl.debug(`[CA] Build Summary: ${buildSummaryLines}`);
-
         if (buildSummaryLines.length > 0) {
             buildSummaryLines.push('');
             buildSummaryLines.push('Code analysis results can be found in the \'Artifacts\' tab.');
         }
 
-        return buildSummaryLines.join('  \r\n');
+        var buildSummaryString = buildSummaryLines.join('  \r\n');
+        tl.debug(`[CA] Build Summary: ${buildSummaryString}`);
```

Comment:
Keeping consistent logging. Now we log the entirety of the build summary rather than the individual lines.